### ISSUE TITLE
Issue installing pg

### ISSUE THREAD

**Issue by mmurad** [larskanis/pg-ldap-sync#6]
Hi,

First off thank you VERY much for this. I am trying to set this up and running into an issue.  I have RHEL 7.1 install with EDB PostgreSQL Plus Advanced server version 9.5.  Everything is running fine on the server as far as the database is concerned.  I want to install pg-ldap-sync on the same server to test it out.  I can install pg-ldap-sync just fine, however, when I attempt to install pg using gem, it gives me an error that really doesn't say much (at least not to me):

My command:  sudo gem install pg -- --with-pg-config=/db/PostgresPlus/9.5AS/bin/pg_config
Result: Building native extensions with: '--with-pg-config=/db/PostgresPlus/9.5AS/bin/pg_config'
This could take a while...
ERROR:  Error installing pg:
    ERROR: Failed to build gem native extension.

```
/usr/bin/ruby extconf.rb --with-pg-config=/db/PostgresPlus/9.5AS/bin/pg_config
```

Using config values from /db/PostgresPlus/9.5AS/bin/pg_config
**\* extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
    --with-opt-dir
         .
         .
    --without-pg-lib=${pg-dir}/
/usr/share/ruby/mkmf.rb:434:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
You have to install development tools first.
    from /usr/share/ruby/mkmf.rb:519:in`try_link0'
    from /usr/share/ruby/mkmf.rb:534:in `try_link'
    from extconf.rb:36:in`<main>'

Gem files will remain installed in /usr/local/share/gems/gems/pg-0.18.4 for inspection.
Results logged to /usr/local/share/gems/gems/pg-0.18.4/ext/gem_make.out

I searched around quite a bit and installed devel packages but to no success.  I then gave up on pg, and decided to install postgres-pr instead.  That install ran fine without error.  I configured my yaml file and attempted to run a test using the following command:

My command: sudo ./pg_ldap_sync -c /etc/pg_ldap_sync.yaml -vv -t
Result: I, [2016-03-29T16:29:02.197334 #27801]  INFO -- : found user-dn: uid=MM7071, ou=People,dc=mykronos,dc=internal
I, [2016-03-29T16:29:02.197726 #27801]  INFO -- : found user-dn: uid=NK7072, ou=People,dc=mykronos,dc=internal
/usr/local/share/gems/gems/postgres-pr-0.7.0/lib/postgres-pr/postgres-compat.rb:16:in `initialize': wrong number of arguments (1 for 7) (ArgumentError)
    from /usr/local/share/gems/gems/pg-ldap-sync-0.1.1/lib/pg_ldap_sync/application.rb:317:in`new'
    from /usr/local/share/gems/gems/pg-ldap-sync-0.1.1/lib/pg_ldap_sync/application.rb:317:in `start!'
    from /usr/local/share/gems/gems/pg-ldap-sync-0.1.1/lib/pg_ldap_sync/application.rb:353:in`run'
    from /usr/local/share/gems/gems/pg-ldap-sync-0.1.1/bin/pg_ldap_sync:6:in `<top (required)>'
    from ./pg_ldap_sync:23:in`load'
    from ./pg_ldap_sync:23:in `<main>'

As you can see, it is connecting to the LDAP server but it is having issues connecting to the database using the connector.  The values in the yaml are all valid.  So now I am stuck.  I can't install pg and I can't get postgres-pr to connect.

Any ideas on how to proceed?

Thanks in advance!

-Mayan

**Comment by larskanis** [comment]
postgres-pr is broken and currently not compatible with pg-ldap-sync. For installation of pg please have a look into the mkmf.log. It should give an indication about what's missing. I guess that either ruby or postgres development files are missing.

**Comment by mmurad** [comment]
Thanks for the quick reply!  I removed postgres-pr and went back to trying to install pg.  The mkmf.log didn't include any error unfortunately, the only thing I saw there was the gcc command.  At that point, I decided to just check the gcc version installed by default on RHEL 7.1 and to my surprise, gcc was missing altogether.  I installed gcc using:

sudo yum install gcc

After doing this, pg installed successfully!  

Thanks again!

-Mayan
